src/commands/setup-ci.ts

```diff
@@ -128,18 +133,17 @@ const runReactNativeCiCli = async (toolbox: CycliToolbox) => {
     }
   }
 
-  const context: ProjectContext = toolbox.projectContext.obtain()
-  toolbox.interactive.surveyStep('Obtained project context.')
+  validateProject(toolbox)
 
-  const snapshotBefore = await toolbox.diff.gitStatus(context)
+  const snapshotBefore = await toolbox.diff.gitStatus()
   toolbox.interactive.surveyStep(
     'Created snapshot of project state before execution.'
   )
 
-  context.selectedOptions = await getSelectedOptions(toolbox)
+  toolbox.context.selectedOptions = await getSelectedOptions(toolbox)
```

Comment:
I'd prefer a setter here and possibly in other places getter instead of just toolbox.context.variable -> so noone is tempted to write (override) something

Reply:
I was never really happy with the fact that we kept `selectedOptions` in context, I now created new small extension `ConfigExtension` which manages the cli configuration - my idea is that it will be the information about what recipes should be generated for what branches, what vendor etc.

For now our config is very simple, it's just a subset of available recipes. So basically I extracted the layer obtaining this information (from flags in case of `--preset` provided and otherwise from survey) to `ConfigExtension`

Now you have to use `toolbox.config.selectedRecipes()` instead of `toolbox.context.selectedOptions`, so also no way to accidentally override the config

13f45f29a7087738cf8e316dc4b7280dad3855df